### ISSUE TITLE
Unexpected proxy object

### ISSUE THREAD

**Issue by moltob** [aranega/pyecore#10]
While implementing the Python generator I am loading the library model from examples as sample model. For some reason I get a proxy object for the `EInt` type. I don't understand why...

This code reproduces the error, shown here as the failing assertion in the end:

```python
def test__load_issue():
    rset = ResourceSet()
    resource = rset.get_resource(URI('../../examples/library.ecore'))
    library_model = resource.contents[0]
    rset.metamodel_registry[library_model.nsURI] = library_model

    book_class = library_model.getEClassifier('Book')
    pages_feature = book_class.findEStructuralFeature('pages')

    assert not isinstance(pages_feature.eType, EProxy)
```

Am I missing something during the load of the model?

Thanks!

**Comment by aranega** [comment]
Thanks for the snipet Mike! At first sight, the code you have seems fine to me, so I'm going to investigate a little bit. I think it should be due to the fact that I automatically build a proxy for `EReferences` perhaps I missed some spot and a proxy is also build for `EAttributes`.

If my guess is right, the solution should be trivial. Stay tune!

**Comment by aranega** [comment]
Ok, I got it, indeed, a proxy is created. Proxy are always created when an `EReference` is read from an XMI file which points to an object from another object which is not contained in a `Resource` (if I recall correctly). So `EInt` and Ecore datatypes are not contained in a resource, so that's why there is a proxy here. I'm going to see what I can do for this, but it should not be an issue. 

However, there is another issue which prevent you to use the proxy as it should be (in a transparent mode), the `EInt` type is not register in the Ecore representation (see `L.1233-1240` from `ecore.py), I totally forgot it! I will add that asap on the develop branch and release a new version (probably 0.3.0, as there is some new stuffs in there),

I try to do this today, asap! For the moment, you can ignore the fact that a proxy is created and you can register the `EInt` type manually (I'm on the fix).

**Comment by moltob** [comment]
Wow... Have you been waiting for a ticket from me? :-) Thanks!

**Comment by aranega** [comment]
I was working a little bit on the project ;-) (I'm trying to implement the command system from EMF on a dedicated branch, everything sounds good at the moment, but the `DeleteCommand` bothers me a little). 

Anyway, I fixed the missing registration, but I cannot anything for the EProxy creation, only because, that's the way the `.ecore` I added to as example is built. The logic behing `EProxy` in PyEcore is quite simple, far more simpler than the 'contained-in-a-resource' thing actually. Each time an `href` is found in an XMI, an `EProxy` towards the pointed element in the resource is created. In the `library.ecore`, each `eType` reference is defined as an `href` (I used GenMyModel to built it), so that's why a proxy is created. To avoid that, the `eType` of each `EStructuralFeature` in the `.ecore` should have been defined using the `xsi:type` tag.

To sum up, the EProxy is not an issue, but the fact that the datatype were not registered were definitely one! I push this on develop and prepare a release!

Thanks again Mike for the snippet and the 'issue'!

**Comment by aranega** [comment]
So, I released a new version on pypi: 0.3.0, you can pull/merge from develop and/or master and your issue should be gone (except the EProxy part which is normal, regarding how the `.ecore` file is defined).

I close the 'issue' by the same occasion ;)
